pkg/machine/e2e/machine_test.go

```diff
@@ -62,10 +62,16 @@ var _ = BeforeSuite(func() {
 	if testProvider.VMType() == define.WSLVirt {
 		pullError = pullWSLDisk()
 	} else {
-		pullError = pullOCITestDisk(tmpDir, testProvider.VMType())
+		// This is a one-off and a little messy but once WSL switches
+		// to use OCI disk artifacts, we can make all the conditionals cleaner.
+		testDiskProvider := testProvider.VMType()
+		if testDiskProvider == define.LibKrun {
+			testDiskProvider = define.AppleHvVirt // libkrun uses the applehv image for testing
+		}
```

Comment:
Mhh, I don't actually see something similar in the actual machine code? How does podman machine init with lkrun know how to pull the image?

Reply:
That is built into the logic at pull time in init; but here we are short-circuiting things by pre-pulling the image for CI.  If you like, I am happy to find the exact spot.

Reply:
its fine, I guess long term we would like to use a different image for krun?

Or maybe even if they are the same image today it would make sense to have to manifest entries for both hypervisors pointing to the same image?